packer-scripts/minimize

```diff
@@ -1,38 +1,47 @@
 #!/usr/bin/env bash
-
 set -o errexit
-set -o xtrace
 
-if [[ $PACKER_BUILDER_TYPE = googlecompute || $PACKER_BUILDER_TYPE = amazon-ebs ]]; then
-  echo "Skipping minimization on $PACKER_BUILDER_TYPE"
-  exit 0
-fi
+main() {
+  set -o xtrace
 
-if [[ $PACKER_BUILDER_TYPE != docker ]]; then
-  readonly SWAP_UUID=$(/sbin/blkid -o value -l -s UUID -t TYPE=swap)
+  if [[ "x${PACKER_BUILDER_TYPE}" = xgooglecompute || "x${PACKER_BUILDER_TYPE}" = xamazon-ebs ]]; then
+    echo "Skipping minimization on ${PACKER_BUILDER_TYPE}"
+    exit 0
+  fi
 
-  if [[ $SWAP_UUID ]]; then
-    readonly SWAP_PART=$(readlink -f /dev/disk/by-uuid/"$SWAP_UUID")
-    /sbin/swapoff "$SWAP_PART"
-    dd if=/dev/zero of="$SWAP_PART" bs=1M || echo "dd exit $? suppressed"
-    /sbin/mkswap -U "$SWAP_UUID" "$SWAP_PART"
+  if [[ "x${PACKER_BUILDER_TYPE}" != xdocker ]]; then
+    readonly SWAP_UUID
+    SWAP_UUID="$(/sbin/blkid -o value -l -s UUID -t TYPE=swap)"
+
+    if [[ "${SWAP_UUID}" ]]; then
+      readonly SWAP_PART
+      SWAP_PART="$(readlink -f "/dev/disk/by-uuid/${SWAP_UUID}")"
+      /sbin/swapoff "${SWAP_PART}"
+      dd if=/dev/zero of="${SWAP_PART}" bs=1M \
+        || echo "dd exit $? suppressed"
+      /sbin/mkswap -U "${SWAP_UUID}" "${SWAP_PART}"
+    fi
   fi
-fi
 
-dd if=/dev/zero of=/EMPTY bs=1M &
-export DD_PID="$!"
-echo 'snoozing a bit here for dd mkay'
-sleep 5
+  dd if=/dev/zero of=/EMPTY bs=1M &
```

Comment:
Wherps should have commented in reply here.  This is the meat of the "minimization", forcing defrag and resulting in a significantly smaller docker image.  The reasoning for doing this hasn't been verified in some time, probably since before docker 1.5, so worth re-verifying given how much time it adds to each packer build.